src/sympc/tensor/mpc_tensor.py

```diff
@@ -1125,6 +1130,32 @@ def ne(self, other: "MPCTensor") -> "MPCTensor":
         other = self.__check_or_convert(other, self.session)
         return 1 - self.eq(other)
 
+    def xor(self, other) -> "MPCTensor":
+        """XOR operator.
+
+        Args:
+            other (MPCTensor): MPCTensor to find xor.
+
+        Returns:
+            MPCTensor: Result of the xor.
+
+        Raises:
+            ValueError : If share class is invalid..
+        """
+        session = self.session
+        op = getattr(operator, "xor")
+        from sympc.tensor import ReplicatedSharedTensor
+
+        if session.protocol.share_class == ReplicatedSharedTensor:
+            shares = [op(share, other) for share in self.share_ptrs]
```

Comment:
We could this split this part between public and private ops, if we incorporate the above changes.